pages/learn/deploy/offline-updates.md

```diff
@@ -0,0 +1,273 @@
+
+```bash
+$ commit=$(balena app ${app_slug} | grep COMMIT | awk '{print $2}')
+
+$ balena preload ${tmpimg} \
+    --app ${app_slug} \
+    --commit ${commit} \
+    --pin-device-to-release
+```
+
+### Create Update media
+
+Next, a USB flash drive, an SD card or any relevant storage media needs to be prepared and flashed. This storage media will be used to update the device through the offline process. Plug in your storage media and run the following the commands.
+
+> Note: The process of flashing will delete all data stored on the storage media. 
+
+```bash
+# list the devices available to flash
+$ balena util available-drives
+/dev/...
+
+# Please make sure `--drive` is really the disk you want to write to. 
+$ sudo balena local flash ${tmpimg} \
+    --drive /dev/{{flash-usb-disk}}
+
+$ rm ${tmpimg}
+```
+
+### Process of reprovisioning 
+
+> Warning: For devices with internal storage this procedure erases the internal media of your device. Hence, remove any mass storage devices containing user data.
+
+With the update media ready having the latest release of the application preloaded. Follow the device's provisioning instructions present on balenaCloud dashboard for your specific device. 
+
+For example: For Raspberrypi devices, insert the recently flashed SD card and power up the device. When the process is complete, (re)connect any mass storage devices containing user data back to the device. Reconnect the device to the local air-gapped network(s). Later, use SSH to connect and inspect application logs, etc.
+
+
+#### Strategies to remotely update with a SD card or USB device
+
+If a device isn’t local, you can just ship the SD card or USB stick/drive to a remote location and let someone there run the update by simply inserting it into the device and booting.
+
+If the target device exists on an air-gapped or Internet restricted network, inserting [ssh](https://www.balena.io/docs/reference/OS/configuration/#sshkeys) keys during the configuration step will allow fleet managers at the remote site to connect into the device directly via OpenSSH and verify the update by examining container logs, etc.
+
+In the event the target device is connected via a low-bandwidth connection, it should eventually establish a connection to balenaCloud, and depending on the quality of the connection, it may respond to Web Terminal commands and output container logs to the dashboard.
+
+### Update Device Registration(s)
+
+This section deals with device registration.
+
+>  manually reflect the new state of the device in balenaCloud for offline devices or devices on private networks without Internet access
+
+#### [Optional] patch state endpoint
+
+This step is strictly not required for devices with low bandwidth internet connections as those devices would be able to update the state endpoint automatically.
```

Comment:
Let me also add that for devices with low bandwidth internet connections, the user has probably already pinned the device to the release it was running before, so that the device doesn't try to start downloading any new release they create.
Given that, it might actually make sense to also set the `should_be_running__release` field in the script below.
What do you think @ab77 ?